### PR TITLE
[test] Add the diskann testcase

### PR DESCRIPTION
**Pull request by jingkl**
Signed-off-by: jingkl <jingjing.jia@zilliz.com>
issue: #19572

### CODE REVIEW

File: tests/python_client/testcases/test_index.py [hunk 1]
```diff
@@ -1547,6 +1548,261 @@ def test_index_with_string_field_empty(self):
...
+        expected: return search success
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data(default_nb)
+        collection_w.insert(data=data)
+
+        def build(collection_w):
+
+            collection_w.create_index(ct.default_float_vec_field_name, default_index_params)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:1798]: Here default_index_params is not diskann index?

File: tests/python_client/testcases/test_index.py [hunk 2]
```diff
@@ -1547,6 +1548,261 @@ def test_index_with_string_field_empty(self):
...
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        index, _ = self.index_wrap.init_index(collection_w.collection, default_float_vec_field_name, ct.default_diskann_index)
+        log.info(self.index_wrap.params)
+        cf.assert_equal_index(index, collection_w.indexes[0])
+        collection_w.load()
+        assert collection_w.num_entities == default_nb
+        vectors = [[random.random() for _ in range(default_dim)] for _ in range(default_nq)]
+        search_res, _=collection_w.search(vectors[:default_nq], default_search_field,
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:1586]: Here should format this line.

And why not use search result check function directly?

File: tests/python_client/testcases/test_index.py [hunk 3]
```diff
@@ -1547,6 +1548,261 @@ def test_index_with_string_field_empty(self):
...
+        res, _ = collection_w.create_index(ct.default_float_vec_field_name, ct.default_diskann_index,
+                                           index_name=ct.default_index_name, _async=_async,
+                                           _callback=self.call_back())
+
+        if _async:
+            res.done()
+            assert len(collection_w.indexes) == 1
+        collection_w.load()
+        vectors = [[random.random() for _ in range(default_dim)] for _ in range(default_nq)]
+        search_res, _=collection_w.search(vectors[:default_nq], default_search_field,
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:1633]: Here should format this line.

And why not use search result check function directly?

File: tests/python_client/testcases/test_index.py (hunk 1 above)
Comment (jingkl) [reply tests/python_client/testcases/test_index.py]: Formatting has been changed

File: tests/python_client/testcases/test_index.py (hunk 3 above)
Comment (jingkl) [reply tests/python_client/testcases/test_index.py]: Formatting has been changed

File: tests/python_client/testcases/test_index.py (hunk 2 above)
Comment (jingkl) [reply tests/python_client/testcases/test_index.py]: Already modified

**Pull request merged**